### PR TITLE
Handle duplicate routes in a graceful manner

### PR DESCRIPTION
**Pull request by prabhataravind**
* With VNET routes, the same route can be learned via multiple sources, like via BGP. Handle this special situation gracefully without orchagent crash during both create and remove operations.
* Add a test case to simulate duplicate route creation/removal

**What I did**

**Why I did it**

**How I verified it**

**Details if related**

### CODE REVIEW

File: tests/test_vnet.py [hunk 1]
```diff
@@ -2478,6 +2532,79 @@ def test_vnet_orch_18(self, dvs, testlog):
...
+        # add another route for same prefix as vnet route
+        dvs.runcmd("vtysh -c \"configure terminal\" -c \"ip route 100.100.1.0/24 10.10.10.3\"")
+
+        # check application database
+        self.pdb.wait_for_entry("ROUTE_TABLE", "100.100.1.0/24")
+
+        # check ASIC route database
+        self.check_route_entries(["100.100.1.0/24"])
+
+        log_string = "Encountered failure in create operation, exiting orchagent, SAI API: SAI_API_ROUTE, status: SAI_STATUS_NOT_EXECUTED"
```
Comment (prsunny) [inline tests/test_vnet.py:2590]: Please add a comment that this log should not be present

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
@prabhataravind , you may need to add a test coverage for delete case

**Comment by prsunny** [comment]
202012 PR - #2666

**Review by prsunny — APPROVED** [review]

**Comment by puffc** [comment]
@yxieca Would you please push this commit to 202211 branch? We're experiencing the duplicate routes problem in 202211. Thank you!

**Comment by puffc** [comment]
@yxieca Thank you for your time.

File: orchagent/saihelper.cpp [hunk 2]
```diff
@@ -612,16 +630,38 @@ task_process_status handleSaiRemoveStatus(sai_api_t api, sai_status_t status, vo
      *          in each orch.
      *       3. Take the type of sai api into consideration.
      */
-    switch (status)
-    {
-        case SAI_STATUS_SUCCESS:
-            SWSS_LOG_WARN("SAI_STATUS_SUCCESS is not expected in handleSaiRemoveStatus");
-            return task_success;
+switch (api)
+{
```
Comment (lguohan) [inline orchagent/saihelper.cpp:634]: identation issue.
